packages/catalog-readmodel-writer/test/catalog-readmodel-writer.integration.test.ts

```diff
@@ -42,42 +67,632 @@ describe("database test", async () => {
     await startedMongoDBContainer.stop();
   });
 
-  describe("Handle message for eservice creation", () => {
-    it("should create an eservice", async () => {
-      const id = generateId();
-      const newEService: EServiceAddedV1 = {
-        eservice: {
-          id,
-          producerId: generateId(),
-          name: "name",
-          description: "description",
-          technology: EServiceTechnologyV1.REST,
-          descriptors: [],
-          createdAt: BigInt(new Date().getTime()),
-          mode: EServiceModeV1.RECEIVE,
-          riskAnalysis: [],
-        },
+  describe("Events V1", async () => {
+    const mockEService = getMockEService();
+
+    describe("EServiceAdded", () => {
+      it("should create an eservice", async () => {
+        const payload: EServiceAddedV1 = {
+          eservice: toEServiceV1(mockEService),
+        };
+        const message: EServiceEventEnvelope = {
+          sequence_num: 1,
+          stream_id: mockEService.id,
+          version: 1,
+          type: "EServiceAdded",
+          event_version: 1,
+          data: payload,
+        };
+        await handleMessageV1(message, eservices);
+
+        const retrievedEservice = await eservices.findOne({
+          "data.id": mockEService.id,
+        });
+
+        expect(retrievedEservice?.data).toEqual(
+          toReadModelEService(mockEService)
+        );
+        expect(retrievedEservice?.metadata).toEqual({ version: 1 });
+      });
+    });
+
+    describe("ClonedEServiceAdded", () => {
+      it("should clone an eservice", async () => {
+        await writeInReadmodel<EServiceReadModel>(
+          toReadModelEService(mockEService),
+          eservices,
+          1
+        );
+        const payload: ClonedEServiceAddedV1 = {
+          eservice: toEServiceV1(mockEService),
+        };
+        const message: EServiceEventEnvelope = {
+          sequence_num: 1,
+          stream_id: mockEService.id,
+          version: 1,
+          type: "ClonedEServiceAdded",
+          event_version: 1,
+          data: payload,
+        };
+        await handleMessageV1(message, eservices);
+
+        const retrievedEservice = await eservices.findOne({
+          "data.id": mockEService.id,
```

Comment:
By searching for this id aren't you retrieving the original eservice instead of the cloned one? Maybe we should find it by its name as defined [here](https://github.com/pagopa/interop-be-monorepo/blob/98916ea82dba01b26a82fc9b27afa4feb7af0561/packages/catalog-process/src/services/catalogService.ts#L1113)

Reply:
Thanks! I didn't notice it. Let me know if the updated solution sounds good.